python/ray/autoscaler/_private/monitor.py

```diff
@@ -309,6 +319,15 @@ def update_load_metrics(self):
                     ip = node_id.hex()
             else:
                 ip = resource_message.node_manager_address
+
+            idle_duration_ms = 0.0
+            if node_id in ray_nodes_idle_duration_ms_by_id:
+                idle_duration_ms = ray_nodes_idle_duration_ms_by_id[node_id]
+            else:
+                logger.warning(
```

Comment:
In which cases will this condition occur?

Reply:
We may need to understand how severe the inconsistency is to determine whether this warrants a warning or a panic. @rickyyx Would you mind providing some insights?

Reply:
Both seem to be getting data from GCS, which should be mostly consistent I believe, but yeah, the codepaths that generate the data is different -> i think graceful handling with warnings is fine (they should eventually be consistent) 

BTW, is there any reason we don't use mainly the v2's info from `get_cluster_resource_state` here entirely? I think the only v1 bit of info that's missing in v2 is "cluster_full", which we could probably also add to v2. 

I don't have a strong opinion between the below given the source of info is both GCS, and the likelihood of inconsistency is low IMO:
1. use mostly v1's info, and patch with v2's idle info
2. use mostly v2's info, and patch with v1's cluster_full or other missing ones in v2. 

I think merging this PR to fix the idle issue is fine - and we could follow up to bring v2's RPC in parity with V1 so we could use V2 entirely here. Or if we are pushing V2 really hard, we might just deprecate V1 in the future.

Reply:
> 1. use mostly v1's info, and patch with v2's idle info
> 2. use mostly v2's info, and patch with v1's cluster_full or other missing ones in v2.

How about we go with option 1 so that we can reduce the dependency between V1 and V2?